kernel/thread.c

```diff
@@ -448,6 +491,20 @@ void _k_thread_single_abort(struct k_thread *thread)
 		}
 	}
 	_mark_thread_as_dead(thread);
+
+#ifdef CONFIG_USERSPACE
+	/* Clear initailized state so that this thread object may be re-used
+	 * and triggers errors if API calls are made on it from user threads
```

Comment:
Shouldn't this happen within `_mark_thread_as_dead()`?

Reply:
what's the advantage of doing it there?
that would increase the scope of what _mark_thread_as_dead() does, currently it is an inline function, only called from here, that marks the thread as _THREAD_DEAD and nothing else.

Reply:
What else calls _mark_thread_as_dead()? Shouldn't this kind of cleanup happens everywhere that's called as well?

Reply:
nothing else calls it
it's an inline function in ksched.h which just sets the dead bit

Reply:
If nothing else calls it, and it's an internal function, might be good to get rid of it and set the bit in this function instead.